guardian/media-atom-maker
# Regenerate youtubePartner API client lib; upgrade to google-api-client 2.x

## What does this change?

Previous attempt to pull in an updated pan-domain-auth #1123 failed because it introduced an updated google-api-client at 2.x. Normally this wouldn't be a problem - we'd update MAM's usage of google clients also to 2.x to resolve the incompatibility and move on with our lives.

Unfortunately MAM uses the semi-private youtubePartner API, which doesn't have a published library. In the past, Google would generate one and host it on their docs pages, but some time in the past few years they've stopped doing so, which would threaten our ability to upgrade to newer versions of the APIs.

Fortunately, google API clients are generated from specifications by scripts - the script is available publicly on github, the spec is available publicly on their APIs, which means we can generate our own! This PR adds a script/docker image to do exactly that, then commits the output from running today, alongside upgrading all of the other Google client libraries to make sure we're in sync, as well as reintroducing #1123 

## How to test

Deploy to CODE, and make sure to do various operations which exercise the various google APIs - get an auth token from panda, view some previously uploaded videos, make some edits, upload a new video etc.

## Have we considered potential risks?

This is only indrectly documented by google - some links are here:
https://developers.google.com/discovery/v1/building-a-client-library
https://github.com/googleapis/google-api-java-client-services/tree/main/generator
